code-conventions.md

```diff
@@ -62,13 +62,21 @@ code. Contributors should also see the dedicated
 * When there are multiple annotations, place them in order:
   - @Override -> @Nullable -> @SuppressWarnings
   - For other annotations, doesn't matter; let your IDE decide
-  
 * When extending one of following classes: SimpleExpression, SimplePropertyExpression, Effect, Condition...
   - Put overridden methods in order
   - SimpleExpression: init -> get (getAll) -> (acceptChange) -> (change) -> isSingle -> getReturnType -> toString
   - SimplePropertyExpression: -> (init) -> convert -> (acceptChange) -> (change) -> getReturnType -> getPropertyName 
   - Effect: init ->  execute  -> toString
   - Condition: init -> check -> toString
+* Return an empty array of the returning type in an expression rather than null.
```

Comment:
Maybe we should make some sort of notice here about it being less supported in the future. Maybe it would be better for `Expression` documentation?

Reply:
It actually already exists in Expression apparently lol

Reply:
Should we still keep this if we are removing the `@Nullable` annotation for the get method?

Reply:
Yes

Reply:
You still have to document what to return.

Reply:
In that case it should be done in the javadocs of the method, not in the code conventions. This isn't just for Skript, it's for addons too

Reply:
It's already in the javadoc as said above

Reply:
All it says is `An array of values for this event. May not contain nulls.`, you should make it clear that if the impl wants to return null for getSingle (i.e. return no values), it must return an array of length 0.

The code conventions change can be reverted, it is not the right place.

Reply:
Seems that this is a very specific case for Expression so I don't see a reason for it to be in Skript's code convetions as the team stated above. Having it in the javadocs is enough.